backend/protocol_rpc/endpoints.py

```
@@ -415,6 +446,168 @@ def count_validators(validators_registry: ValidatorsRegistry) -> int:
+                        data={"contract_address": contract_address},
+                    )
+                if signer.lower() != deployer.lower():
+                    raise JSONRPCError(
+                        code=-32000,
+                        message="Only contract deployer can upgrade",
+                        data={"signer": signer, "deployer": deployer},
+                    )
+                # Authorized - proceed with upgrade
+            except JSONRPCError:
+                raise
+            except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)